### PR TITLE
[PETSc] 3.20.5

### PR DESCRIPTION
**Pull request by boriskaus**
This compiles PETSc version 3.20.5. 
Compared to previous efforts, we have fewer external dependencies, which will make this hopefully more maintainable. It comes with `MUMPS` and `SuperLU_Dist` on linux/mac. MPI on windows remains broken so is deactivated.

### CODE REVIEW

File: P/PETSc/build_tarballs.jl [hunk 1]
```diff
@@ -1,35 +1,42 @@
...
-MUMPS_COMPAT_VERSION = "5.6.2"
-SUPERLUDIST_COMPAT_VERSION = "8.1.2"   
-MPItrampoline_compat_version="5.2.1"
+version = v"3.20.5"
+petsc_version = v"3.20.5"
+SUPERLUDIST_COMPAT_VERSION = "8.2.1"   
 BLASTRAMPOLINE_COMPAT_VERSION="5.8.0"    
+SUITESPARSE_COMPAT_VERSION = "7.7"       
 
+MUMPS_COMPAT_VERSION = "5.6.2"
```
Comment (amontoison) [inline P/PETSc/build_tarballs.jl:13]: Is it still needed?
You don't depend anymore on `MUMPS_jll.jl` or `MUMPS_seq_jll.jl`.
For information, The latest release is `5.7.3`.

Comment (boriskaus) [reply P/PETSc/build_tarballs.jl]: thanks for pointing this out - I have removed the compat definition where this is no longer necessary

**Comment by giordano** [comment]
Why this is now failing?

**Comment by boriskaus** [comment]
Looks like I messed up the merge from master - will look at it on Wednesday

**Pull request merged**